bugwarrior/services/gitlab.py

```diff
@@ -500,71 +567,58 @@ def issues(self):
                 rid = repo['id']
-                merge_requests.update(
-                    self.get_repo_merge_requests(rid)
+                self.repo_map[rid] = repo
+                issues.update(
+                    self.get_repo_issues(rid)
                 )
-            log.debug(" Found %i merge requests.", len(merge_requests))
-            merge_requests = list(filter(self.include, merge_requests.values()))
-            log.debug(" Pruned down to %i merge requests.", len(merge_requests))
-
-            yield from self._get_issue_objs(merge_requests,
-                                              'merge_request',
-                                              repo_map)
-
-        if self.config.include_todos:
-            todos = self.get_todos()
-            log.debug(" Found %i todo items.", len(todos))
-            if not self.config.include_all_todos:
-                todos = list(filter(self.include_todo(repos), todos))
-            log.debug(" Pruned down to %i todos.", len(todos))
-
-            for project, todo in todos:
-                if project is not None:
-                    repo = project
-                else:
-                    repo = {
-                        'path': 'the instance',
-                    }
-                todo['repo'] = repo['path']
-
-                todo_obj = self.get_issue_for_record(todo)
-                todo_url = todo['target_url']
-                projectName = repo['path']
-                if self.config.project_owner_prefix:
-                    projectName = repo['namespace']['path'] + "." + projectName
-                extra = {
-                    'issue_url': todo_url,
-                    'project': projectName,
-                    'namespace': "todo",
-                    'type': 'todo',
-                    'annotations': [],
-                }
-                todo_obj.update_extra(extra)
-                yield todo_obj
+            log.debug(" Found %i issues.", len(issues))
+            issues = list(filter(self.include, issues.values()))
+            log.debug(" Pruned down to %i issues.", len(issues))
+
+            yield from self._get_issue_objs(issues, 'issue')
+
+            if not self.config.filter_merge_requests:
+                merge_requests = {}
+                for repo in repos:
+                    rid = repo['id']
+                    merge_requests.update(
+                        self.get_repo_merge_requests(rid)
+                    )
+                log.debug(" Found %i merge requests.", len(merge_requests))
+                merge_requests = list(filter(self.include, merge_requests.values()))
+                log.debug(" Pruned down to %i merge requests.", len(merge_requests))
+
+                yield from self._get_issue_objs(merge_requests,
+                                                  'merge_request')
+
+            if self.config.include_todos:
+                yield from self.yield_todos_from_query('todos?state=pending')
```

Comment:
Sorry, I think I was wrong and todos should be top level here so that this option will still work if the query is for issues or merge requests. Perhaps we should add a validator that throws an error if `item_type == 'todo' and self.config.include_todos`.

Reply:
I'll have another look